env0/resource_template.go

```diff
@@ -440,33 +421,24 @@ func templateCreatePayloadFromParameters(prefix string, d *schema.ResourceData)
 }
 
 // Reads template and writes to the resource data.
-func templateRead(prefix string, template client.Template, d *schema.ResourceData, isImport bool) error {
+func templateRead(prefix string, template client.Template, d *schema.ResourceData) error {
 	pathPrefix := "path"
 	terragruntTfBinaryPrefix := "terragrunt_tf_binary"
-	terraformVersionPrefix := "terraform_version"
 
 	if prefix != "" {
-		pathPrefix = prefix + ".0." + pathPrefix
 		terragruntTfBinaryPrefix = prefix + ".0." + terragruntTfBinaryPrefix
-		terraformVersionPrefix = prefix + ".0." + terraformVersionPrefix
+		pathPrefix = prefix + ".0." + pathPrefix
 	}
 
 	path, pathOk := d.GetOk(pathPrefix)
-	terragruntTfBinary := d.Get(terragruntTfBinaryPrefix).(string)
-	terraformVersion := d.Get(terraformVersionPrefix).(string)
-
-	// If this value isn't set, ignore whatever is returned from the response.
-	// This helps avoid drifts when defaulting to 'opentofu' for new 'terragrunt' templates, and 'terraform' for existing 'terragrunt' templates.
-	// 'template.TerragruntTfBinary' field is set to 'omitempty'. Therefore, the state isn't modified if `template.TerragruntTfBinary` is an empty string.
-	// This is not true for imports - because the shcema is empty irrespective in that case.
-	if !isImport {
-		if terragruntTfBinary == "" {
+
+	// This is done to avoid drifts in case the backend returns "opentofu", but non is configured in the provider.
+	// (The provider implicitly defaults to "opentofu").
+	if template.TerragruntTfBinary == client.OPENTOFU {
```

Comment:
1. But this PR will do a breaking change on purpose, so why do we need that? 
2. if BE returns "opentofu" and non is configured in the provider, it means this template was "terraform" before. So we should actually set "terraform", or actually show the drift. Am I missing something? i

Reply:
If the BE returns opentofu - then it's opentofu.
The backend is the source of truth.

However, if nothing is set in the provider - then - the defaul is "opentofu".
For various hashicorp reasons, I can't set default as "opentofu" in the terraform schema. But still want avoid the drift for the "opentofu" case. `'' -> 'opentofu'`.

If in the backend the value is "terraform" - and nothing was set in the provider - the user will get a drift message `terraform -> null`. To avoid the drift they will have to explicitly add `terragrunt_tf_binary = 'terraform'`

Now this is assuming that the backend (source of truth) will always return the correct value.

Reply:
If the BE returns opentofu - then it's opentofu.
The backend is the source of truth.

However, if nothing is set in the provider - then - the defaul is "opentofu".
For various hashicorp reasons, I can't set default as "opentofu" in the terraform schema. But still want avoid the drift for the "opentofu" case. `'' -> 'opentofu'`.

If in the backend the value is "terraform" - and nothing was set in the provider - the user will get a drift message `terraform -> null`. To avoid the drift they will have to explicitly add `terragrunt_tf_binary = 'terraform'`

Now this is assuming that the backend (source of truth) will always return the correct value.

Reply:
> terraform -> null

isn't it `'' -> 'terraform'`? I didnt get this part

and thanks for the explanation

Reply:
yeah - I think you're right. Just flip the examples.